src/app/auth/service/auth.register.service.ts

```diff
@@ -17,8 +17,10 @@ export default class AuthRegisterService
   constructor(private userRepository: Repository<User>) {}
 
   async call(params: AuthRegisterDto): Promise<User> {
-    this.ensureUserNotExist(params.email)
-
+    const isUserExist = await this.ensureUserNotExist(params.email)
+    if (isUserExist) {
+      throw new UserDuplicatedEmailException('A User with that email already exists')
+    }
```

Comment:
@xyingsoft 
please look at above changes.
ensureUserNotExist is async function, but had a call without any await, or promise handler.
And if that function is throwing, we need to add catcher ?
So I have changed this line like above.